packages/core/src/stylable-resolver.ts

```diff
@@ -287,63 +297,100 @@ export class StylableResolver {
     public resolveElement(meta: StylableMeta, symbol: StylableSymbol) {
         return this.resolveName(meta, symbol, true);
     }
-    public resolveParts(meta: StylableMeta, diagnostics: Diagnostics) {
-        const resolvedClasses: Record<string, Array<CSSResolve<ClassSymbol | ElementSymbol>>> = {};
-        for (const [className, classSymbol] of Object.entries(meta.getAllClasses())) {
-            resolvedClasses[className] = this.resolveExtends(
-                meta,
-                className,
-                false,
-                undefined,
-                (res, extend) => {
-                    const decl = findRule(meta.ast, '.' + className);
-                    if (decl) {
-                        // ToDo: move to STExtends
-                        if (res && res._kind === 'js') {
-                            diagnostics.error(decl, CSSClass.diagnostics.CANNOT_EXTEND_JS(), {
-                                word: decl.value,
-                            });
-                        } else if (res && !res.symbol) {
-                            diagnostics.error(
-                                decl,
-                                CSSClass.diagnostics.CANNOT_EXTEND_UNKNOWN_SYMBOL(extend.name),
-                                { word: decl.value }
-                            );
-                        } else {
-                            diagnostics.error(decl, CSSClass.diagnostics.IMPORT_ISNT_EXTENDABLE(), {
-                                word: decl.value,
-                            });
-                        }
-                    } else {
-                        if (classSymbol.alias) {
-                            meta.ast.walkRules(new RegExp('\\.' + className), (rule) => {
-                                diagnostics.error(
-                                    rule,
-                                    CSSClass.diagnostics.UNKNOWN_IMPORT_ALIAS(className),
-                                    { word: className }
-                                );
-                                return false;
-                            });
-                        }
+
+    public resolveSymbols(meta: StylableMeta, diagnostics: Diagnostics) {
+        const resolvedSymbols: MetaResolvedSymbols = {
+            mainNamespace: {},
+            class: {},
+            element: {},
+            var: {},
+            js: {},
+            keyframes: {},
+            cssVar: {},
+            import: {},
+        };
+        // resolve main namespace
+        for (const [name, symbol] of Object.entries(meta.getAllSymbols())) {
+            let deepResolved: CSSResolve | JSResolve | null;
+            if (symbol._kind === `import` || (symbol._kind === `cssVar` && symbol.alias)) {
+                deepResolved = this.deepResolve(symbol);
+                if (!deepResolved || !deepResolved.symbol) {
+                    // ToDo: handle...
```

Comment:
unresolved imports are reported as part of the `validateImports` process at the beginning of transformation, it might be possible to unify this process, but I'm not sure about duplication of reports for `mixins/value()` and I prefer to wait with it for now.

I added a comment instead of the ToDo.